providers/redis.js

```diff
@@ -0,0 +1,69 @@
+const { Provider } = require('klasa');
+const { Client } = require('redis-nextra');
+
+module.exports = class extends Provider {
+
+	constructor(...args) {
+		super(...args, { enabled: true });
+
+		const { hosts, options } = this.client.providers.redis;
+		this.db = new Client(hosts, options);
+
+		this.db.on('ready', () => this.client.emit('debug', 'Redis initialized.'))
+			.on('serverReconnect', server => this.client.emit('warn', `Redis server: ${server.host.string} is reconnecting`))
+			.on('error', err => this.client.emit('err', err));
+	}
+
+	hasTable(table) {
+		return this.db.tables.has(table);
+	}
+
+	createTable(table) {
+		return this.db.createTable(table);
+	}
+
+	deleteTable(table) {
+		return this.db.deleteTable(table);
+	}
+
+	getAll(table) {
+		return this.db.table(table).valuesJson('*');
+	}
+
+	get(table, id) {
+		return this.db.table(table).getJson(id);
+	}
+
+	has(table, id) {
+		return this.db.table(table).has(id);
+	}
+
+	getRandom(table) {
+		return this.db.table(table).srandmember();
+	}
+
+	create(table, id, data) {
+		return this.db.table(table).setJson(id, data);
+	}
+
+	set(...args) {
+		return this.create(...args);
+	}
+
+	insert(...args) {
+		return this.create(...args);
+	}
+
+	update(...args) {
+		return this.set(...args);
```

Comment:
This should get and merge the current document to the current value.

Reply:
Redis already overwrites the value if it exists.

Reply:
That's why, it's update, not replace.

Reply:
I don't understand.

Reply:
Something like this:

https://github.com/dirigeants/klasa/blob/dbd1375e4dd235788281605f65b115f948e36ee6/src/providers/json.js#L150-L151